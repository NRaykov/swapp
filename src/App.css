.App-header {
  width: 100%;
  margin-bottom: 2rem;
  background-color: #000000;
}
.logo {
  font-size: 3rem;
  font-family: 'StarWars';
}
.loginButton {
  width: 3.5rem;
  height: 3.5rem;
  border-radius: 0.5rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #ffffff;
}
.loginButton svg {
  color: #72d5dc;
}

.loginButton:hover {
  background-color: #f8e35a;
}
.loginButton:hover svg {
  color: white;
}

