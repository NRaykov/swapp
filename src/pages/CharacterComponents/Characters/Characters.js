import React from 'react';
import { useQuery } from '@apollo/react-hooks';
import {charactersQuery} from "../../../client/queries";

import {CardImg, Col, Row} from "reactstrap";
import { Link } from 'react-router-dom';
import Button from "../../../components/StylesComponents/elements/button/button";
import styles from "./styles.module.css";
import Card from "../../../components/StylesComponents/elements/card/card";
import Heading from "../../../components/StylesComponents/elements/heading/heading"
import ErrorHandler from "../../../components/LoginSystem/guards/errorHandler";


const Characters = () => {

  const placeholderImage = "https://i.etsystatic.com/17236199/r/il/598fda/1553216794/il_570xN.1553216794_ayds.jpg";

  let first = 12;
  const { data, loading, error, fetchMore } = useQuery(charactersQuery, {
    variables: { first }
  });

  if (loading) return 'Loading ...';
  if (error)return (<ErrorHandler/>);

  const [...allPeople] = data.allPeople.edges;
  let { hasNextPage, endCursor } = data.allPeople.pageInfo;

  const loadMore = () => {
    fetchMore({
      variables: {
        after: endCursor,
        first: 12
      },
      updateQuery: (prev, { fetchMoreResult: { allPeople } }) => {

        return {
          allPeople: {
            ...allPeople,
            edges: [
              ...prev.allPeople.edges,
              ...allPeople.edges]
          },
        };
      },
    });
  };
        const characters = allPeople.map(({ node }) => (
            <Col md="4" key={node.id}>
              <Link className={ styles.cardPanel }
                    variant="nav" key={node.id}
                    to={`/characters/${node.id}`}>
                <Card variant="primary" className={styles.cardPanel}>
                  <CardImg src={node.image === null ? placeholderImage : node.image} className={`${styles.thumbnail} img-fluid foo`} alt="Card image cap" />
                  <div className={`${styles.cardBody}`}>
                    <Heading variant="primary" className={styles.primaryHeading}>{node.name}</Heading>
                  </div>
                </Card>
              </Link>
            </Col>
        ));




  return (
          <React.Fragment>
              <Row>
                {characters}
              </Row>
              {hasNextPage && (

              <Col md="12">
                <div className={`${styles.buttonPanel}`}>
                  <Button variant="primary" className={`${styles.btnLogin}`} onClick={loadMore}>Load More</Button>
                </div>
              </Col>
              )}
          </React.Fragment>



  );
};
export default Characters;

