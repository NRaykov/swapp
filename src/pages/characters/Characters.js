import React from 'react';
import Login from '../../components/Login/Login.js';
import { isAuthenticated, signout } from '../../guards/auth.js';
import { withRouter } from 'react-router-dom';

const Characters = withRouter(({ history }) =>
            isAuthenticated ? (
                    <React.Fragment>
                        <h1>Characters</h1>
                    </React.Fragment>

            ) : ( <Login/> ),
        );
export default Characters;
