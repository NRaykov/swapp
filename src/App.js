import React, {useState} from 'react';
//Apollo Boost
import { ApolloProvider } from '@apollo/react-hooks';
import ApolloClient from "apollo-client"

import { InMemoryCache } from 'apollo-cache-inmemory';
import { createHttpLink } from 'apollo-link-http';

import authLink from './client/auth';
import {typeDefs} from './client/local'

//Navigation and Login
import { BrowserRouter } from 'react-router-dom';
import Container from './components/StylesComponents/elements/container/container';
import Header from "./components/Header/Header";
import Home from "./pages/HomeComponent/Home";
//Styling
import './App.css';
import { ThemeProvider } from 'styled-components';
import ThemeChanger from './components/StylesComponents/themeChanger';
import { themes } from './components/StylesComponents';

    const httpLink = createHttpLink({
      uri: 'https://swapp.st6.io/graphql',
    });

    const cache = new InMemoryCache();

    //Export 'client' to use it outside of this scope
    export const client = new ApolloClient({
      cache,
      link: authLink.concat(httpLink),
      resolvers: {
          /** TODO  Create Resolvers Here **/
      },
      typeDefs,
    });

    const isAuthed = !!localStorage.getItem('token');
    cache.writeData({
      data: {
        authenticated: isAuthed,
      },
    });


export const AppComponent = () => {

    /**** Set Theme to LocalStorage ****/
    const [theme, setTheme] = useState('light');
    const themeChanger = ThemeChanger(theme, setTheme);

    localStorage.setItem('Theme', 'light');
    const getTheme = localStorage.getItem('Theme');
    console.log(getTheme);


  return (
        <ApolloProvider client={client}>
            <BrowserRouter>
              <ThemeProvider theme={themes[getTheme]}>
                <Container variant="primary" className="container-main">
                    <Header themeChanger={themeChanger}/>
                    <div className="container py-5">
                      <Home themeChanger={themeChanger}/>
                    </div>
                </Container>
              </ThemeProvider>
            </BrowserRouter>
        </ApolloProvider>
  )
};

