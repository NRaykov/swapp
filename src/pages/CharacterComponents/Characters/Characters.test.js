import React from 'react';
import Characters from './Characters'
import { ApolloProvider } from "react-apollo";
import '../../../setupTests';
import { client } from '../../../App'
const ReactTestRenderer = require('react-test-renderer');

describe('Snapshot test "Characters" Component', () => {

    it('Should compare the component with a snapshot', () => {
      const component = ReactTestRenderer.create(
        <ApolloProvider client={client}>
          <Characters />
        </ApolloProvider>
      );
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    })
  });
