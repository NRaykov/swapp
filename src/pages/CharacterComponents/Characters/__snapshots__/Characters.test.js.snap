// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test "Characters" Component Should compare the component with a snapshot 1`] = `"Loading ..."`;

exports[`Snapshot test Should compare the component with a snapshot 1`] = `"Loading ..."`;
